install/migrations/update_9.5.x_to_10.0.0/cable.php

```diff
@@ -0,0 +1,232 @@
+      `cabletypes_id` int NOT NULL DEFAULT '0',
+      `comment` text,
+      `date_mod` timestamp NULL DEFAULT NULL,
+      `date_creation` timestamp NULL DEFAULT NULL,
+      PRIMARY KEY (`id`),
+      KEY `name` (`name`),
+      KEY `item_endpoint_a` (`itemtype_endpoint_a`,`items_id_endpoint_a`),
+      KEY `item_endpoint_b` (`itemtype_endpoint_b`,`items_id_endpoint_b`),
+      KEY `items_id_endpoint_b` (`items_id_endpoint_b`),
+      KEY `items_id_endpoint_a` (`items_id_endpoint_a`),
+      KEY `socketmodels_id_endpoint_a` (`socketmodels_id_endpoint_a`),
+      KEY `socketmodels_id_endpoint_b` (`socketmodels_id_endpoint_b`),
+      KEY `sockets_id_endpoint_a` (`sockets_id_endpoint_a`),
+      KEY `sockets_id_endpoint_b` (`sockets_id_endpoint_b`),
+      KEY `cablestrands_id` (`cablestrands_id`),
+      KEY `states_id` (`states_id`),
+      KEY `complete` (`entities_id`,`name`),
+      KEY `is_recursive` (`is_recursive`),
+      KEY `users_id_tech` (`users_id_tech`),
+      KEY `cabletypes_id` (`cabletypes_id`),
+      KEY `date_mod` (`date_mod`),
+      KEY `date_creation` (`date_creation`)
+    ) ENGINE=InnoDB DEFAULT CHARSET = {$default_charset} COLLATE = {$default_collation} ROW_FORMAT=DYNAMIC;";
+   $DB->queryOrDie($query, "10.0 add table glpi_cables");
+
+   $migration->addField('glpi_states', 'is_visible_cable', 'bool', [
+      'value' => 1,
+      'after' => 'is_visible_appliance'
+   ]);
+   $migration->addKey('glpi_states', 'is_visible_cable');
+}
+
+if (!$DB->tableExists('glpi_sockets') && $DB->tableExists('glpi_netpoints')) {
+
+   //create socket table
+   $query = "CREATE TABLE `glpi_sockets` (
+      `id` int NOT NULL AUTO_INCREMENT,
+      `position` int NOT NULL DEFAULT '0',
+      `entities_id` int NOT NULL DEFAULT '0',
+      `is_recursive` tinyint NOT NULL DEFAULT '0',
+      `locations_id` int NOT NULL DEFAULT '0',
+      `name` varchar(255) DEFAULT NULL,
+      `socketmodels_id` int NOT NULL DEFAULT '0',
+      `wiring_side` tinyint DEFAULT '1',
+      `itemtype` varchar(255) DEFAULT NULL,
+      `items_id` int NOT NULL DEFAULT '0',
+      `networkports_id` int NOT NULL DEFAULT '0',
+      `comment` text,
+      `date_mod` timestamp NULL DEFAULT NULL,
+      `date_creation` timestamp NULL DEFAULT NULL,
+      PRIMARY KEY (`id`),
+      KEY `name` (`name`),
+      KEY `socketmodels_id` (`socketmodels_id`),
+      KEY `complete` (`entities_id`,`locations_id`,`name`),
+      KEY `is_recursive` (`is_recursive`),
+      KEY `location_name` (`locations_id`,`name`),
+      KEY `item` (`itemtype`,`items_id`),
+      KEY `networkports_id` (`networkports_id`),
+      KEY `wiring_side` (`wiring_side`),
+      KEY `date_mod` (`date_mod`),
+      KEY `date_creation` (`date_creation`)
+    ) ENGINE=InnoDB DEFAULT CHARSET = {$default_charset} COLLATE = {$default_collation} ROW_FORMAT=DYNAMIC;";
+   $DB->queryOrDie($query, "10.0 add table glpi_sockets");
+
+
+   //migrate link between NetworkPort and Socket
```

Comment:
Maybe I am wrong, but maybe it could be simplier to directly create a socket for each existing NetworkPort, instead of fetching every NetworkPortEthernet and NetworkPortFiberchannel to get its parent NetworkPort.